### PR TITLE
Add .gitignore Vagrantfile and requirements.txt

### PR DESCRIPTION
**Pull request by OliverZWang**
This PR closes Issue #4. 

- Add `.gitignore` file
- Add `requirements.txt`
- Add `Vagrantfile`

### CODE REVIEW

**Comment by pkutlesa** [comment]
Thanks for taking initiative and making the first PR Oliver! It looks good. I want to point out that there may be some unnecessary lines in the ```.gitignore``` file. For example, I don't think we'll be using scrapy.

**Comment by OliverZWang** [comment]
> Thanks for taking initiative and making the first PR Oliver! It looks good. I want to point out that there may be some unnecessary lines in the `.gitignore` file. For example, I don't think we'll be using scrapy.

@pkutlesa Thank you for the comment. I removed the lines for `Scrapy` and `Django` as you mentioned. Feel free to let me know if there are other directories that are not necessary to include and I will remove them as well. 

Also, it might not harmful to include extra lines in the `.gitinore` file (within reason, of course), but we can certainly clean it up wherever we can. 

If it looks okay to you, feel free to approve so that we can merge this PR and unblock other issues. Thank you.

**Comment by luke0zong** [comment]
Great work!
We might want to update the dependencies to the latest commit as in https://github.com/nyu-devops/lab-flask-tdd/blob/master/requirements.txt

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Runtime dependencies
+Flask==1.1.1
```
Comment (luke0zong) [inline requirements.txt:2]: Flask==1.1.2

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+Flask-API==1.1
+Flask-SQLAlchemy==2.4.1
```
Comment (luke0zong) [inline requirements.txt:4]: Flask-SQLAlchemy==2.4.4

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+python-dotenv==0.10.3
+psycopg2-binary==2.8.4
+
+# Testing
+nose==1.3.7
+rednose==1.3.0
+pinocchio==0.4.2
+coverage==4.5.4
+pylint>=2.4.1
```
Comment (luke0zong) [inline requirements.txt:13]: pylint==2.4.4
flake8==3.7.9

**Review by luke0zong — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 3 above)
Comment (OliverZWang) [reply requirements.txt]: Done. Thank you, @luke0zong : )

**Review by luke0zong — APPROVED** [review]

**Pull request merged**